olive/engine/engine.py

```diff
@@ -207,6 +208,7 @@ def run(
         self,
         input_model: OliveModel,
         packaging_config: Optional[PackagingConfig] = None,
+        aml_client: Optional[AzureMLClient] = None,
```

Comment:
do we really need to make the aml_client concept known to engine? 

Ideally, only model.py should download the model when the model is loaded.

Reply:
Another use case for aml_client for Engine will be AML packaging. Engine will need to have aml_client to package out models to AML with AML_client. This part of logic is not sent as a PR for now but will be needed in the near future.

Reply:
could we make the aml_client part of the packaging logic instead of engine?